### DISCUSSION TITLE
Does NextJS remove duplicate imports across files?

### DISCUSSION THREAD

**Discussion by michael-mcgreal** [vercel/next.js#40524] [Help] [answered]
For example, when I import Next/Link – my VsCode extension shows the 'import cost' as 71.5K (gzipped: 22.3K). If I import this across many components, it can add up (especially for larger imports).

Does NextJS remove duplicate imports like this? I think the answer is yes. Maybe in a process like webpack (possibly via tree-shaking)?

Any advice on more specific research keywords would be much appreciated!

_In some cases, to avoid duplicate imports, I've made a single file for imports- i.e. utils/types (for prisma generated types)- and then import them from there. But I'm starting to think that this is more harmful, and can interfere with processes like code-splitting (if NextJS does remove duplicate imports)._

**Comment by gaston-flores** [comment, marked as answer]
Yes, Next handles tree-shaking/code-splitting for you. The import cost is a one time fee, feel free to import `next/link` as many times as you like 😉 

If you'd like to see what Next is putting in your bundles try [@next/bundle-analyzer](https://www.npmjs.com/package/@next/bundle-analyzer). It generates a nice graph with all generated files, their size, and which modules are in each one.

**Reply by michael-mcgreal** [reply]
That's superb – Many thanks, gaston!
